### ISSUE TITLE
[web][safari][html] Image color is not applied

### ISSUE THREAD

**Issue by hussamDana92** [flutter/flutter#91578]
When I import an image on flutter and use the color attribute, it's color doesn't change. It works fine on desktop and browser android but not in browser IOS.
Steps to reproduce:
1 - import an image using Image widget pure white color .
2 - set the color of the image as gray

Sample code:

```dart
import 'package:flutter/material.dart';
void main() {
 runApp(App());
}

class App extends StatelessWidget {
 @override
 Widget build(BuildContext context) {
    return Image.asset("assets/image.png", color: Colors.gray);
 }
}
```
Expected(with gray color):
<img width="109" alt="b38de960-2654-4b36-b8c8-97f6b19be73c" src="https://user-images.githubusercontent.com/69424680/136693017-370b2737-be69-456c-878f-295e8409f1d6.png">

Actual:
![1287b800-e64b-4eaf-a941-72812725fdf9](https://user-images.githubusercontent.com/69424680/136693029-d3c3edcf-b38d-4675-96b6-162807263940.png)

<details>
  <summary>Logs</summary>

```
```

```
```

```
[✓] Flutter (Channel stable, 2.5.2, on macOS 11.4 20F71 darwin-arm, locale en-KW)
    • Flutter version 2.5.2 at /Users/Developer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3595343e20 (10 days ago), 2021-09-30 12:58:18 -0700
    • Engine revision 6ac856380f
    • Dart version 2.14.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.52.1)
    • VS Code at /Users/Downloads/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (2 available)
    • SM A107F (mobile) • R9AN100CW3J • android-arm    • Android 11 (API 30)
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 94.0.4606.71

• No issues found!

```

</details>

**Comment by danagbemava-nc** [comment]
Hi @hussamDana92, thanks for filing the issue. 

Can you please provide some of the sample images to make it easier for us to reproduce? 

Thank you

**Comment by hussamDana92** [comment]
Hi @danagbemava-nc 

Still the issue is there in Flutter 2.8.0

**Comment by raghadalgwere** [comment]
Still happing in 2.10

**Comment by MOdeh90** [comment]
Still Issue .. we can't change color by this code 
`Image.asset("assets/image.png", color: Colors.gray)`

in safari browser when using the html renderer  (Flutter 2.10.0 Stable)

**Comment by bschmalbSP** [comment]
Still Issue in 2.10.2
.. we can't change color by this code

**Comment by RahmiTufanoglu** [comment]
I can confirm

> Still Issue in 2.10.2 .. we can't change color by this code

**Comment by ajay-prabhakar** [comment]
Tested in flutter version `3.0.0`
Still it persists.

Left is chrome, right one is safari

![image](https://user-images.githubusercontent.com/89897211/168963146-c2c2e03a-bd8f-4003-92bc-b21ead52efa6.png)

**Comment by uniqmuz** [comment]
Still issue in 3.0.1
I think it's related with BlendMode.
in iOS safari, it seems other filter applied with BlendMode.dstATop.
I changed to BlendMode.darken as workaround in my case.

**Comment by liandrade** [comment]
Still issue in 3.0.2
`Image.asset(
      'images/trevo.png',
      color: Colors.deepPurple,
      width: size,
    );`

The image stay with white color.

**Comment by VinhBui0508** [comment]
The same Error

**Comment by LucaIaconelli** [comment]
Any update on this? The error is still here with flutter 3.3.3

**Comment by MarcBritton** [comment]
Still seen with Flutter 3.3.4

**Comment by subash1327** [comment]
for me the image is white  in iPad safari and chrome on the latest version of flutter

**Comment by danReynolds** [comment]
> Still issue in 3.0.1 I think it's related with BlendMode. in iOS safari, it seems other filter applied with BlendMode.dstATop. I changed to BlendMode.darken as workaround in my case.

I don't think this blend mode works on Flutter web with the html renderer yea. Pretty bad bug imo

**Comment by Naing-Linn-6802** [comment]
Any update on this bug ?

**Comment by rsercano** [comment]
Any update? Still happening

**Comment by jsmeke** [comment]
Same issue here.

**Comment by dimityrivanov** [comment]
Same issue in 3.10.6

**Comment by YumNumm** [comment]
Same issue in 3.13.4 😢

**Comment by duldiev** [comment]
Let me update that it is not working still

**Comment by Ysr7** [comment]
I still have a problem, any solution?

**Comment by Ysr7** [comment]
I got the solution by changing from **flutter build web --release --web-renderer html** to **Flutter build web --release --web-renderer canvaskit**

CanvasKit renderer
This renderer is fully consistent with Flutter for mobile and desktop, has faster performance with higher widget density, but adds about 1.5 MB in download size. CanvasKit uses WebGL to render Skia painting commands.

**Comment by mym0404** [comment]
@Ysr7 Read issue. This issue is covering html renderer

**Comment by zhpoo** [comment]
any update?

**Comment by rahulrmishra** [comment]
Any updates?

**Comment by dimityrivanov** [comment]
You can use the SVG library to render images at any colours you like so this issue is no biggie or we can just change the icon to whatever color we need it

**Comment by guilhermedaldim** [comment]
Any update? The error is still in 3.19.5

**Comment by Jashemaits** [comment]
for anyone looking for solution you can use ColorFiltered like this 
```
ColorFiltered(
      colorFilter: ColorFilter.mode(
        color, // Replace with your desired color
        BlendMode.srcATop,
      ),
      child: image,
    )
```

**Comment by yjbanov** [comment]
As the HTML renderer has been deprecated and removed from the default Flutter SDK, we're now closing all issues that affect only the HTML renderer. For more information, please see:

* https://github.com/flutter/flutter/issues/145954
* https://github.com/flutter/flutter/pull/164003
